collector/src/main.rs

```diff
@@ -249,57 +253,87 @@ fn bench(
 
     let start = Instant::now();
     let mut skipped = false;
-    for (nth_benchmark, benchmark) in benchmarks.iter().enumerate() {
-        let is_fresh =
-            rt.block_on(conn.collector_start_step(artifact_row_id, &benchmark.name.to_string()));
+
+    let mut measure_and_record = |benchmark_name: &execute::BenchmarkName,
+                                  benchmark_supports_stable: bool,
+                                  print_intro: &dyn Fn(),
+                                  measure: &dyn Fn(
+        &mut execute::BenchProcessor,
+    ) -> Result<(), anyhow::Error>| {
```

Comment:
This formatting seems pretty ungreat -- maybe we can convince rustfmt to do this better? I can't seem to trivially reproduce on playground in any case...

Reply:
Changing `Result<(), anyhow::Error>` to `anyhow::Result<()>` is enough.